quipucords/tests/scanner/openshift/test_inspect.py

```diff
@@ -250,3 +257,45 @@ def test_inspect_errors_on_extra_cluster_facts(
             "message": teapot_response.data.decode(),
         }
     }
+
+
+@pytest.mark.django_db
+def test_inspect_with_enabled_workloads(  # noqa: PLR0913
+    mocker,
+    scan_task: ScanTask,
+    cluster,
+    node_ok,
+    workload,
+    operators,
+    use_dummy_feature_flag,
+):
+    """Test connecting to OpenShift host with enabled workloads."""
+    mocker.patch.object(OpenShiftApi, "retrieve_cluster", return_value=cluster)
+    mocker.patch.object(OpenShiftApi, "retrieve_nodes", return_value=[node_ok])
+    mocker.patch.object(OpenShiftApi, "retrieve_workloads", return_value=[workload])
+    mocker.patch.object(OpenShiftApi, "retrieve_operators", return_value=operators)
+
+    runner = InspectTaskRunner(scan_task=scan_task, scan_job=scan_task.job)
+    runner.execute_task(mocker.Mock())
+
+    raw_facts = scan_task.get_facts()
+    cluster = raw_facts[-1]
+    assert "workloads" in cluster.keys()
+
+
+@pytest.mark.django_db
+def test_inspect_with_disabled_workloads(  # noqa: PLR0913
+    mocker, scan_task: ScanTask, cluster, node_ok, operators
+):
+    """Test connecting to OpenShift host with workloads default behavior."""
+    mocker.patch.object(OpenShiftApi, "retrieve_cluster", return_value=cluster)
+    mocker.patch.object(OpenShiftApi, "retrieve_nodes", return_value=[node_ok])
+    mocker.patch.object(OpenShiftApi, "retrieve_operators", return_value=operators)
```

Comment:
could you also mock "retrieve_operators" method and check it was not called after execute_task is executed?

Reply:
retrieve_workloads u mean? sure thing, I will add that check.

Reply:
yep, that haha